### PR TITLE
Add a test for failure when using v4 signatures.

### PR DESCRIPTION
**Pull request by timuralp**
Adds a unit test with the Amazon Java SDK for failures when submitting
v4-signed requests.

### CODE REVIEW

File: src/test/java/org/gaul/s3proxy/S3AwsSdkTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    @Test
+    public void testAwsV4Failure() throws Exception {
+        BasicAWSCredentials awsCreds = new BasicAWSCredentials(s3Identity,
+                s3Credential);
+        AmazonS3 client = new AmazonS3Client(awsCreds);
+        client.setEndpoint(s3Endpoint.toString());
+        try {
+            client.getObject(containerName, "foo");
```
Comment (andrewgaul) [inline src/test/java/org/gaul/s3proxy/S3AwsSdkTest.java]: You need to call `fail()` call here to ensure that `S3Proxy` throws the expected exception.

**Comment by timuralp** [comment]
Updated to use the ExpectedException rule.

File: src/test/java/org/gaul/s3proxy/S3AwsSdkTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            s3Context.close();
+        }
+        if (context != null) {
+            context.getBlobStore().deleteContainer(containerName);
+            context.close();
+        }
+    }
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
```
Comment (andrewgaul) [inline src/test/java/org/gaul/s3proxy/S3AwsSdkTest.java]: If you rename this to `thrown` it will ignore the visibility.

**Comment by andrewgaul** [comment]
Commit message should mention that we use the AWS SDK instead of jclouds due to [JCLOUDS-480](https://issues.apache.org/jira/browse/JCLOUDS-480).

**Comment by andrewgaul** [comment]
Please address Checkstyle nits then we can merge this.

**Pull request merged**